bugsnag-android-core/src/main/java/com/bugsnag/android/Configuration.java

```diff
@@ -0,0 +1,821 @@
+     * return false; // ignore the session
+     * }
+     * })
+     *
+     * @param onSession a callback to run before a session is captured
+     * @see OnSessionCallback
+     */
+    @Override
+    public void addOnSession(@NonNull OnSessionCallback onSession) {
+        if (onSession != null) {
+            impl.addOnSession(onSession);
+        } else {
+            error("addOnSession");
+        }
+    }
+
+    /**
+     * Removes a previously added "on session" callback
+     * @param onSession the callback to remove
+     */
+    @Override
+    public void removeOnSession(@NonNull OnSessionCallback onSession) {
+        if (onSession != null) {
+            impl.removeOnSession(onSession);
+        } else {
+            error("removeOnSession");
+        }
+    }
+
+    /**
+     * Adds a map of multiple metadata key-value pairs to the specified section.
+     */
+    @Override
+    public void addMetadata(@NonNull String section, @NonNull Map<String, ?> value) {
+        if (section != null && value != null) {
+            impl.addMetadata(section, value);
+        } else {
+            error("addMetadata");
+        }
+    }
+
+    /**
+     * Adds the specified key and value in the specified section. The value can be of
+     * any primitive type or a collection such as a map, set or array.
+     */
+    @Override
+    public void addMetadata(@NonNull String section, @NonNull String key, @Nullable Object value) {
+        if (section != null && key != null) {
+            impl.addMetadata(section, key, value);
+        } else {
+            error("addMetadata");
+        }
+    }
+
+    /**
+     * Removes all the data from the specified section.
+     */
+    @Override
+    public void clearMetadata(@NonNull String section) {
+        if (section != null) {
+            impl.clearMetadata(section);
+        } else {
+            error("clearMetadata");
+        }
+    }
+
+    /**
+     * Removes data with the specified key from the specified section.
+     */
+    @Override
+    public void clearMetadata(@NonNull String section, @NonNull String key) {
+        if (section != null && key != null) {
+            impl.clearMetadata(section, key);
+        } else {
+            error("clearMetadata");
+        }
+    }
+
+    /**
+     * Returns a map of data in the specified section.
+     */
+    @Nullable
+    @Override
+    public Map<String, Object> getMetadata(@NonNull String section) {
+        if (section != null) {
+            return impl.getMetadata(section);
+        } else {
+            error("getMetadata");
+            return null;
+        }
+    }
+
+    /**
+     * Returns the value of the specified key in the specified section.
+     */
+    @Nullable
+    @Override
+    public Object getMetadata(@NonNull String section, @NonNull String key) {
+        if (section != null && key != null) {
+            return impl.getMetadata(section, key);
+        } else {
+            error("getMetadata");
```

Comment:
I don't think logging makes sense for an accessor method.

Reply:
As above, we're validating values passed to all of our public API methods, whether they are setters, getters, or otherwise.